WordPress/Classes/Utility/Migrator/CoreDataIterativeMigrator.swift

```diff
@@ -0,0 +1,272 @@
+            throw error
+        }
+
+        return backupURL
+    }
+
+    /// Copy migrated over the original files
+    ///
+    static func copyMigratedOverOriginal(from tempDestinationURL: URL, to storeURL: URL) throws {
+        do {
+            let fileManager = FileManager.default
+            let files = try fileManager.contentsOfDirectory(atPath: tempDestinationURL.deletingLastPathComponent().path)
+            try files.forEach { (file) in
+                if file.hasPrefix(tempDestinationURL.lastPathComponent) {
+                    let fullPath = tempDestinationURL.deletingLastPathComponent().appendingPathComponent(file).path
+                    let toPath = storeURL.deletingLastPathComponent().appendingPathComponent(file).path
+                    try? fileManager.removeItem(atPath: toPath)
+                    try fileManager.moveItem(atPath: fullPath, toPath: toPath)
+                }
+            }
+        } catch {
+            DDLogError("⛔️ Error while copying migrated over the original files: \(error)")
+            throw error
+        }
+    }
+
+    /// Delete backup copies of the original file before migration
+    ///
+    static func deleteBackupCopies(at backupURL: URL) throws {
+        do {
+            let fileManager = FileManager.default
+            let files = try fileManager.contentsOfDirectory(atPath: backupURL.path)
+            try files.forEach { (file) in
+                let fullPath = URL(fileURLWithPath: backupURL.path).appendingPathComponent(file).path
+                try fileManager.removeItem(atPath: fullPath)
+            }
+        } catch {
+            DDLogError("⛔️ Error while deleting backup copies of the original file before migration: \(error)")
+            throw error
+        }
+    }
+}
+
+
+// MARK: - Private helper functions
+//
+private extension CoreDataIterativeMigrator {
+
+    static func migrateStore(at url: URL,
+                             storeType: String,
+                             fromModel: NSManagedObjectModel,
+                             toModel: NSManagedObjectModel,
+                             with mappingModel: NSMappingModel) -> Bool {
+        let tempDestinationURL = createTemporaryFolder(at: url)
+
+        // Migrate from the source model to the target model using the mapping,
+        // and store the resulting data at the temporary path.
+        let migrator = NSMigrationManager(sourceModel: fromModel, destinationModel: toModel)
+        do {
+            try migrator.migrateStore(from: url,
+                                      sourceType: storeType,
+                                      options: nil,
+                                      with: mappingModel,
+                                      toDestinationURL: tempDestinationURL,
+                                      destinationType: storeType,
+                                      destinationOptions: nil)
+        } catch {
+            return false
```

Comment:
Maybe log the error here, and in the following do/try/catch block, before returning?

Reply:
Never mind. I just saw they are logged where thrown :)